packages/realm/src/Listeners.ts

```diff
@@ -54,3 +56,43 @@ export class Listeners<CallbackType> {
     this.listeners.clear();
   }
 }
+
+/** @internal */
+type BindingListenerCallback = (r: Realm, name: string, schema?: Realm.ObjectSchema[]) => void;
+/** @internal */
+export class BindingListeners {
+  /**
+   * Keeps tracked of registered listener callbacks for Realm class notifications.
+   */
+  constructor(private name: string) {}
+  private listeners = new Set<BindingListenerCallback>();
+
+  // Combined callback which runs all listener callbacks in one call.
+  callback(realm: Realm): void {
```

Comment:
Would it make sense for this `realm` to be a private member of the class instead? It will always get the same Realm passed in as callback, we might as well make that explicit.